tests/main/ubuntu-core-snapd-run-from-snap/task.yaml

```diff
@@ -0,0 +1,42 @@
+summary: Check that the snapd.run-from-snap helper works
+
+systems: [ubuntu-core-*]
+
+restore: |
+    # cleanup generated units (just in case stop does not do its job)
+    rm -f /run/systemd/system/snapd.*
+    # cleanup mount
+    umount /snap/snapd/current/usr/lib/snapd || true
+    # and ensure snapd is working (needed in restore)
+    systemctl daemon-reload
+    systemctl start snapd.service snapd.socket
+
+execute: |
+    echo "Install the snapd snap"
+    snap install --edge snapd
+
+    echo "Stop snapd from core"
+    systemctl stop snapd.service snapd.socket
+    systemctl status snapd|MATCH "Active: inactive"
+
+    echo "Ensure our local changes are in the snap"
+    mount -o bind /usr/lib/snapd /snap/snapd/current/usr/lib/snapd/
+
+    echo "Run the manage script (from the snap)"
+    ls /run/systemd/system > before
+    /snap/snapd/current/usr/lib/snapd/snapd.run-from-snap start /snap/snapd/current
+    echo "Check that the snapd from the snap is running"
+    systemctl status snapd|MATCH "Active: active"
+    systemctl status snapd|MATCH "/snap/snapd/current/usr/lib/snapd/snapd"
+
+    echo "And ensure snapd from the snap is working on a basic level"
+    snap list
```

Comment:
Nitpick, can we do "snap version" here and add a TODO to grep/MATCH for a marker (we could say something like origin: snapd.snap, origin: core.snap, origin: classic, etc.